src/ruby/yast/y2start_helpers.rb

```diff
@@ -69,6 +71,33 @@ def self.setup_signals
       end
     end
 
+    # Returns application title string
+    def self.application_title(client_name)
+      # do not fail if gethostname failed
+      hostname = Socket.gethostname rescue ""
+      hostname = "" if hostname == "(none)"
+      hostname = " @ #{hostname}" unless hostname.empty?
+      if Yast::Arch.s390
+        # e.g. stdout "2964 = z13 IBM z13" transfered into "IBM z13"
+        architecture = Yast::Execute.on_target!(
+          "/usr/bin/read_values", "-c").split("=")[1].strip rescue ""
+        arch_array = architecture.split(' ')
+        arch_array.shift if arch_array.size > 1
+        architecture = arch_array.join(' ')
+        if !Yast::UI.TextMode
+          # Show the S390 architecutue in the QT banner only.
+          # The environment variable YAST_BANNER will be read and shown
+          # in libyui-qt.
+          ENV["YAST_BANNER"] = architecture
+          architecture = ""
+        end
+      else
+        architecture = ""
+      end
+      left_title = "YaST2 - #{client_name}#{hostname}"
+      left_title + architecture.rjust(80-left_title.size)
```

Comment:
test case? this is critical part of YaST so I would like to see it very well test covered.